app/src/main/java/com/tpc/nudj/ui/screens/myClubs/MyClubsViewModel.kt

```
@@ -0,0 +1,141 @@
+package com.tpc.nudj.ui.screens.myClubs
+
+import android.util.Log
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.google.firebase.firestore.FirebaseFirestore
+import com.tpc.nudj.model.ClubUser
+import com.tpc.nudj.repository.follow.FollowRepository
+import com.tpc.nudj.repository.user.UserRepository
+import com.tpc.nudj.utils.FirestoreCollections
+import com.tpc.nudj.utils.FirestoreUtils
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.tasks.await
+import javax.inject.Inject
+
+
+@HiltViewModel
+class MyClubsViewModel @Inject constructor(
+    private val userRepository: UserRepository,
+    private val followRepository: FollowRepository
+) : ViewModel() {
```

We want some error handling as well. So we need to propagate the errors the screens and not just log them.
So either extend the viewmodel from BaseViewModel which already have error handling setup. But you will have to create ui state for that, which would be a lot of change to the code.
So for this screen, you have add a separate flow for propagating errors to ui. Collect them in screen and use snackbar do display the error.
But in future try to use BaseViewModel as it reduces some boilerplate code for complex screens.